### PR TITLE
[ LSTM ] Add Skeleton LSTM Layer Class @open sesame 04/01 11:08

### PR DESCRIPTION
**Pull request by jijoongmoon**
This PR includes skeleton code of lstm layer

Resolves:

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

File: nntrainer/layers/lstm.h [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+   */
+  ActiFunc acti_func;
+
+  /**
+   * @brief     activation type for recurrent : default is sigmoid
+   */
+  ActivationType recurrent_activation_type;
+
+  /**
+   * @brief     activation function for recurrent : default is sigmoid
```
Comment (kparichay) [inline nntrainer/layers/lstm.h:136]: Can you add a small comparison detail for where this activation will be used compared to acti_func?

**Review by kparichay — APPROVED** [review]
LGTM other than the minor comment

**Review by zhoonit — APPROVED** [review]
LGTM :)

File: nntrainer/layers/lstm.h (hunk 1 above)
Comment (jijoongmoon) [reply nntrainer/layers/lstm.h:136]: I will add follow up pr. Thanks.

**Pull request merged**